packages/fxa-payments-server/src/routes/Product.tsx

```diff
@@ -0,0 +1,209 @@
+  useEffect(() => {
+    if (accessToken) {
+      dispatch(actions.fetchPlans(accessToken));
+      dispatch(actions.fetchSubscriptions(accessToken));
+    }
+  }, [ dispatch, productId, accessToken ]);
+
+  if (isLoading) {
+    return <LoadingSpinner />;
+  }
+
+  if (plans.error) {
+    return <div>(plans error! {'' + plans.error})</div>;
+  }
+
+  if (subscriptions.error) {
+    return <div>(subscriptions error! {'' + subscriptions.error})</div>;
+  }
+
+  const productPlans = plansByProductId(productId);
+  const selectedPlan = productPlans[planIdx];
+
+  if (! selectedPlan) {
+    return <div>No plans available for this product.</div>;
+  }
+
+  if (createSubscriptionStatus.loading) {
+    return <div>Creating subscription...</div>;
+  }
+
+  if (createSubscriptionStatus.error) {
+    return <div>
+      Problem creating subscription:
+      {'' + createSubscriptionStatus.error}
+    </div>;
+  }
+
+  if (createSubscriptionStatus.result) {
+    return <div>
+      <h2>TODO: Redirect to product goes here</h2>
+      <Link to="/subscriptions">Manage subscriptions</Link>
+    </div>;
+  }
+
+  const alreadyHasProduct = subscriptions.result
+    .some(subscription => subscription.productName === productId);
+  if (alreadyHasProduct) {
+    return <div>
+      <h2>TODO: Already have a subscription to the product. Redirect to product goes here?</h2>
+      <Link to="/subscriptions">Manage subscriptions</Link>
+    </div>;
+  }
+
+  return (
+    <div>
+      <div>
+        <h2>Let&apos;s set up your subscription</h2>
+        {/* TODO: only one plan per product for MVP, but including some plan selection buttons for demo purposes */}
+        {productPlans.length > 1 && (
+          <ul>
+            {/* eslint-disable camelcase */}
+            {productPlans.map(({ plan_id }: Plan, idx: number) => (
+              <li key={plan_id}>
+                <button onClick={() => setPlanIdx(idx)}>Select plan {plan_id}</button>
+              </li>
+            ))}
+            {/* eslint-enable camelcase */}
+          </ul>
+        )}
+        <ProductValueProposition plan={selectedPlan} />
+      </div>
+      <Elements>
+        <SubscriptionForm {...{
+          accessToken,
+          createSubscription,
+          selectedPlan,
+        }} />
+      </Elements>
+    </div>
+  );
+};
+
+type SubscriptionFormProps = {
+  accessToken: string,
+  selectedPlan: Plan,
+  createSubscription: Function,
+};
+export const SubscriptionFormRaw = ({
+  accessToken,
+  selectedPlan,
+  createSubscription,
+  stripe,
+}: SubscriptionFormProps & ReactStripeElements.InjectedStripeProps) => {
+  const [ confirmationChecked, onConfirmationChanged ] = useCheckboxState();
+
+  const onSubmit = useCallback(ev => {
+    ev.preventDefault();
+
+    // TODO: use react state on form fields along with validation
+    const data = new FormData(ev.target);
+    const name = String(data.get('name'));
+
+    if (stripe) {
```

Comment:
Should this also throw if `stripe` is undefined?

Reply:
Maybe? Stripe *should* be defined by virtue of the JS included in the header and passed along in props via injectStripe. But, TypeScript got lost following that. So, the stripe prop is declared as optional (even though it isn't) and code here had to be guarded in a conditional.